examples/xt.lua

```diff
@@ -0,0 +1,51 @@
+local xt = require("xtable")
+local action = xt.action
+local family = xt.family
+local match_packet
+local check_packet
+local destroy_packet
+
+local udp = 0x11
+local dns = 0x35
+
+function match_packet(skb)
+	print("match_packet called\n")
+	-- ip header
+	local ipb = skb:getbyte(0)
+	local ihlen = (( ipb << 4 ) & 0xFF ) >> 4
+	local iphdrlen = ihlen * 4
+	local proto = skb:getbyte(9)
+
+	if proto == udp then
+		local srcport = skb:getbyte(iphdrlen) << 8 | skb:getbyte(iphdrlen + 1)
+		local dstport = skb:getbyte(iphdrlen + 2) << 8 | skb:getbyte(iphdrlen + 3)
+		if srcport == dns or dstport == dns then
```

Comment:
I didn't verify it, but just a reminder that you should take endianess in consideration.. probably it's something that should be added to luadata (as the `const` thing.. probably it would be the case of having `->flags`)

Reply:
We should also have luadata `short` get/set too, there we can handle the endianness.

Reply:
I'm not sure what would be better.. to have separate APIs per byte length (such as {get,set}short) or having the byte length as argument on {get,set}number.. mixed feelings.. what do you think?

Reply:
Both approaches are fine.